File: crates/domain/src/execution.rs
use chrono::{DateTime, Utc};
use easyjob_common::{ExecutionId, TaskId, TriggerId};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
pub enum ExecutionStatus {
    Queued,
    Running,
    Succeeded,
    Failed,
    TimedOut,
    Cancelled,
    Skipped,
    Interrupted,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct Execution {
    pub id: ExecutionId,
    pub task_id: TaskId,
    pub trigger_id: Option<TriggerId>,
    pub status: ExecutionStatus,
    pub scheduled_at: Option<DateTime<Utc>>,
    pub started_at: DateTime<Utc>,
    pub finished_at: Option<DateTime<Utc>>,
    pub duration_ms: Option<u64>,
    pub exit_code: Option<i32>,
    pub error_message: Option<String>,
}

impl Execution {
    pub fn new(
        task_id: TaskId,
        trigger_id: Option<TriggerId>,
        scheduled_at: Option<DateTime<Utc>>,
    ) -> Self {
        Self {
            id: ExecutionId::new(),
            task_id,
            trigger_id,
            status: ExecutionStatus::Queued,
            scheduled_at,
            started_at: Utc::now(),
            finished_at: None,
            duration_ms: None,
            exit_code: None,
            error_message: None,
        }
    }
}

File: apps/desktop/src/types/execution.ts
import type { TaskId, TriggerId } from './task';

export type ExecutionId = string;

export type ExecutionStatus =
  | 'Queued'
  | 'Running'
  | 'Succeeded'
  | 'Failed'
  | 'TimedOut'
  | 'Cancelled'
  | 'Skipped'
  | 'Interrupted';

export function getStatusLabel(status?: string): string {
  switch (status) {
    case 'Running':
      return '运行中';
    case 'Succeeded':
      return '成功';
    case 'Failed':
      return '失败';
    case 'TimedOut':
      return '超时';
    case 'Cancelled':
      return '已取消';
    case 'Queued':
      return '排队中';
    case 'Skipped':
      return '已跳过';
    case 'Interrupted':
      return '异常中断';
    default:
      return status || '-';
  }
}

export function getStatusTagType(
  status?: string
): 'info' | 'success' | 'error' | 'warning' | 'default' {
  switch (status) {
    case 'Running':
      return 'info';
    case 'Succeeded':
      return 'success';
    case 'Failed':
      return 'error';
    case 'TimedOut':
      return 'warning';
    case 'Cancelled':
      return 'default';
    case 'Interrupted':
      return 'warning';
    default:
      return 'default';
  }
}

export interface Execution {
  id: ExecutionId;
  task_id: TaskId;
  trigger_id: TriggerId | null;
  status: ExecutionStatus;
  scheduled_at: string | null;
  started_at: string;
  finished_at: string | null;
  duration_ms: number | null;
  exit_code: number | null;
  error_message: string | null;
}

export interface ExecutionOutputPayload {
  execution_id: ExecutionId;
  task_id: TaskId;
  stream: 'stdout' | 'stderr';
  content: string;
}
